pkg/controller/perconaservermongodb/psmdb_controller.go

```diff
@@ -372,6 +399,21 @@ func (r *ReconcilePerconaServerMongoDB) Reconcile(request reconcile.Request) (re
 		}
 	}
 
+	err = r.deleteMongos(cr)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "delete mongos")
+	}
+
+	err = r.deleteCfg(cr)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "delete config server")
+	}
+
+	err = r.reconcileMongos(cr)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "reconcile Deployment for")
+	}
+
```

Comment:
```go
r.deleteMongos(cr)
...
r.reconcileMongos(cr)
```
I don't know. On first glance, it looks like the mongos being recreated (and configsvr deleted) on each reconcile loop. I see it's not. But it looks a bit confusing

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/439/commits/76e8eff8897a6a1d3d42cca9e4f794c06d7a9aff